internal/sdkv2provider/data_source_tunnel.go

```diff
@@ -0,0 +1,77 @@
+package sdkv2provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareTunnel() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCloudflareTunnelRead,
+
+		Schema: map[string]*schema.Schema{
+			consts.AccountIDSchemaKey: {
+				Description: consts.AccountIDSchemaDescription,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the tunnel",
+				ForceNew:    true,
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the tunnel",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the tunnel. Possible values: inactive, degraded, healthy, down",
+			},
+			"tunnel_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of the tunnel. Possible values: cfd_tunnel, warp_connector",
```

Comment:
similar to https://github.com/cloudflare/terraform-provider-cloudflare/pull/2866/files#r1367994208, we should use the format helper for consistency here.